microservices/users/src/services/change-password.ts

```diff
@@ -104,10 +122,40 @@ class ChangePassword {
 
     user.password = newPassword;
 
-    await this.repository.encryptPassword(user);
+    await Promise.all([this.repository.encryptPassword(user), this.handleClearUserTokens(user.id)]);
 
     return this.repository.save(user);
   }
+
+  /**
+   * Handle clear user tokens
+   */
+  private async handleClearUserTokens(userId: string): Promise<void> {
+    // Check default clear tokens type
+    const { changePasswordClearTokensType } = await remoteConfig();
+
+    const type = this.clearTokensType || changePasswordClearTokensType;
+
+    if (!type || type === 'none') {
+      return;
+    }
+
+    if (type === 'all') {
+      await this.repository.clearUserTokens(userId);
+
+      return;
+    }
+
+    // Clear tokens type - REST. Keep only current
+    if (!this.currentToken) {
+      throw new BaseException({
+        status: 500,
+        message: 'Failed to clear rest user tokens. Current user token was not found.',
+      });
```

Comment:
actualy do we need throw exception? You need to clean up token, but if token not exist, we just skip it